nixos/modules/services/networking/pihole-ftl.nix

```diff
@@ -0,0 +1,107 @@
+{ pkgs
+, lib
+, config
+, ...
+}:
+
+with lib;
+
+let
+
+  cfg = config.services.pihole-ftl;
+  stateDir = "/etc/pihole";
```

Comment:
🚨

When we start calling variables that point to `/etc` by `stateDir` in NixOS the red alert sirens start up. Generally people in the NixOS community aren't a fan of using configuration directories for state.

You have a problem, though. It appears this software is **very** stubborn when it comes to doing almost everything NixOS doesn't want software to do. Just take a look at [this](https://discourse.pi-hole.net/t/what-files-does-pi-hole-use/1684).

Can we use [RootDirectory](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#RootDirectory=) to shoehorn this service into its own space?

Reply:
It is pretty common to edit the config values for pihole over the web interface. That wouldn't really work on nixos.